text/0000-canonical-metadata.md

```diff
@@ -0,0 +1,1208 @@
+    domEvents?: DOMEvent[];
+
+    // Programmatically added event listeners in a module
+    eventListeners?: ProgrammaticEventListener[];
+
+    // EventTarget.dispatchEvent invocations
+    eventsDispatched?: EventDispatch[];
+
+    /**
+     * Static resources referenced in a module.
+     * Includes any fully qualified urls referenced in javascript.
+     * Does not include resources loaded via lightning/platformResourceLoader or a url determined at runtime.
+     */
+    staticResources?: StaticResourceReference[];
+}
+
+// Information about a class
+interface Class {
+    // Is optional because class can be anonymous
+    name?: string;
+
+    // A class is considered a component class if it extends
+    // 'LightningElement' class available in the standard 'lwc' module.
+    isComponentClass?: boolean;
+
+    // A parent can be another class declared locally, imported identifier from an external module.
+    // When not statically analyzable, will be 'unresolved'(e.g. Mixin)
+    extends: ParentClass;
+
+    properties: ClassProperty[];
+
+    methods: ClassMethod[];
+
+    // Location starts with the "class" key word and ends at the closing curly braces of the class body
+    location: SourceLocation;
+
+    // A string literal representing the leading comments of the class declaration
+    doc?: string;
+    id: ID;
+}
+
+interface ClassMember {
+    // Note: getter/setter types are considered a property
+    type: 'property' | 'method';
+    // According to javascript specs
+    propertyFieldType: 'public' | 'private' | 'static';
+    name: string;
+    // Location starts with the first character of the member's identifier and ends at the semi-colon
+    // in case of a property and closing curly in case of a method
+    // For a accessor property type, the getter and setter property has the specific location
+    location?: SourceLocation;
+    // A string literal representing the leading comments of the class member
+    doc?: string;
+}
+
+// Information about a class property of a component class.
+interface ClassProperty extends ClassMember {
+    type: 'property';
+    // To make it easy to discover
+    propertyType: 'accessor' | 'dataProperty';
+    // True, if the property is an instance property
+    dataProperty?: DataProperty;
+    // Preset if the property has a getter
+    getter?: Getter;
+    // Present if the property has a setter
+    setter?: Setter;
+    decorators?: [TrackDecorator | WireDecorator | ApiDecorator];
+}
+
+// The naming is derived from data descriptors v/s accessor descriptor
+interface DataProperty {
+    // If the initial value is not statically analyzable, type will be 'unresolved'
+    initialValue: {
+        type:
+            | 'number'
+            | 'string'
+            | 'boolean'
+            | 'null'
+            | 'object'
+            | 'undefined'
+            | 'array'
+            | 'unresolved';
+        value?: any;
```

Comment:
@ravijayaramappa As we discussed today, it would be useful to have the canonical shape for objects and arrays defined for `initialValue.value` and `static` field `value`s.